pkg/export/prom/prom.go

```diff
@@ -1024,6 +1043,11 @@ func (r *metricsReporter) deleteTargetInfo(uid svc.UID, service *svc.Attrs) {
 	r.targetInfo.DeleteLabelValues(targetInfoLabelValues...)
 }
 
+func (r *metricsReporter) deleteSurveyInfo(uid svc.UID, service *svc.Attrs) {
+	targetInfoLabelValues := r.labelValuesTargetInfo(r.origService(uid, service))
+	r.surveyInfo.DeleteLabelValues(targetInfoLabelValues...)
```

Comment:
is there a use case to keep track terminated services? otherwise would better to set 0 instead of deleting

Reply:
I think we want to remove it always, otherwise for example, a pod that's crash looping can create explosion of data in this metric.